src/Parser/EnhancementShaman/Modules/ShamanCore/Pandemic.js

```diff
@@ -0,0 +1,58 @@
+import SPELLS from 'common/SPELLS';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import Module from 'Parser/Core/Module';
+
+const SPELLID = SPELLS.FLAMETONGUE_BUFF.id;
+const PANDEMIC_THRESHOLD = 11500;
+const debug = false;
+
+class Pandemic extends Module {
+  ft_timestamp = [];
+  early_refresh = [];
+  refresh_avearge;
+  static dependencies = {
+    combatants: Combatants,
+  }
+
+  on_byPlayer_applybuff(event) {
+    if(event.ability.guid === SPELLID) {
+      this.ft_timestamp.push(event.timestamp);
+
+    }
+  }
+
+  on_byPlayer_refreshbuff(event) {
+    if(event.ability.guid === SPELLID) {
+      this.ft_timestamp.push(event.timestamp);
+    }
+  }
+
+  on_finished() {
+    let diff;
+
+    for (let i = 1; i < this.ft_timestamp.length; i++) {
+      diff = this.ft_timestamp[i] - this.ft_timestamp[i - 1];
+      if(diff < PANDEMIC_THRESHOLD) {
+        this.early_refresh.push(diff);
+      }
+    }
+    debug && console.log(this.early_refresh);
+    if(this.early_refresh.length > 0) {
+      this.refresh_avearge = this.early_refresh.reduce((a, b) => a + b) / this.early_refresh.length;
```

Comment:
I'd prefer not doing complete calculations in the on_finished method. Since refresh_average is based purely on class properties you could also make refresh_average a getter that calculates it when called.